src/main/java/org/threadly/concurrent/future/watchdog/PollingWatchdog.java

```diff
@@ -0,0 +1,143 @@
+   * (which is lazily constructed).  This should be fine in most cases, but you can provide your 
+   * own scheduler if you have specific needs where the {@link CentralThreadlyPool} default is not 
+   * a good option.
+   * 
+   * @param pollFrequencyMillis Time in milliseconds that futures will be checked if they should be timed out
+   * @param sendInterruptOnFutureCancel If {@code true}, and a thread is provided with the future, 
+   *                                      an interrupt will be sent on timeout
+   */
+  public PollingWatchdog(long pollFrequencyMillis, boolean sendInterruptOnFutureCancel) {
+    this(getStaticScheduler(), pollFrequencyMillis, sendInterruptOnFutureCancel);
+  }
+  
+  /**
+   * Constructs a new {@link PollingWatchdog} with a scheduler of your choosing.  It is critical that 
+   * this scheduler has a free thread available to inspect futures which may not have completed in 
+   * the given timeout.  You may want to use a org.threadly.concurrent.limiter to ensure that 
+   * there are threads available.
+   * 
+   * @param scheduler Scheduler to schedule task to look for expired futures
+   * @param pollFrequencyMillis Time in milliseconds that futures will be checked if they should be timed out
+   * @param sendInterruptOnFutureCancel If {@code true}, and a thread is provided with the future, 
+   *                                      an interrupt will be sent on timeout
+   */
+  @SuppressWarnings("unchecked")
+  public PollingWatchdog(SubmitterScheduler scheduler, long pollFrequencyMillis, 
+                         boolean sendInterruptOnFutureCancel) {
+    super((futures) -> new CheckRunner(scheduler, pollFrequencyMillis, 
+                                       sendInterruptOnFutureCancel, 
+                                       (Collection<PollingFutureWrapper>) futures));
+    
+    // scheduler not null verified in CheckRunner
+    ArgumentVerifier.assertGreaterThanZero(pollFrequencyMillis, "pollFrequencyMillis");
+  }
+
+  /**
+   * Watch a given {@link ListenableFuture} to ensure that it completes within the constructed 
+   * time limit.  If the future is not marked as done by the time limit then it will be completed 
+   * by invoking {@link ListenableFuture#cancel(boolean)}.  Weather a {@code true} or 
+   * {@code false} will be provided to interrupt the running thread is dependent on how this 
+   * {@link PollingWatchdog} was constructed.
+   * <p>
+   * The provided {@link Supplier} will be invoked regularly as the state is polled.  Once it 
+   * returns {@code true} it will never be invoked again, and instead 
+   * {@link ListenableFuture#cancel(boolean)} will be invoked on the provided future. 
+   * 
+   * @param cancelTest Supplier to return {@code true} when the future should be canceled
+   * @param future Future to inspect to ensure completion
+   */
+  public void watch(Supplier<Boolean> cancelTest, ListenableFuture<?> future) {
+    if (future == null || future.isDone()) {
```

Comment:
Seems a little odd, but should we run cancelTest here?

Reply:
I considered running it.  But seems unlikely that you would want a future canceled as soon as you submit it.  I also worried about concerns running in the invoking thread when it's expected to be async.  So thought I would just avoid the call here and wait for the next poll.